android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/AdapterOperations.kt

```diff
@@ -1,20 +1,49 @@
 package com.solana.mobilewalletadapter.clientlib
 
 import android.net.Uri
+import androidx.annotation.Nullable
 import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
 
 interface AdapterOperations {
-    suspend fun authorize(identityUri: Uri, iconUri: Uri, identityName: String, rpcCluster: RpcCluster = RpcCluster.MainnetBeta): MobileWalletAdapterClient.AuthorizationResult
+
+    @Deprecated(
+        "Replaced by updated authorize() method, which adds MWA 2.0 spec support",
+        replaceWith = ReplaceWith("authorize(identityUri, iconUri, identityName, chain, authToken, features, addresses)"),
+        DeprecationLevel.WARNING
+    )
+    suspend fun authorize(
+        identityUri: Uri,
+        iconUri: Uri,
+        identityName: String,
+        rpcCluster: RpcCluster = RpcCluster.MainnetBeta
+    ): MobileWalletAdapterClient.AuthorizationResult
+
+    suspend fun authorize(
+        identityUri: Uri,
+        iconUri: Uri,
+        identityName: String,
+        chain: String,
+        authToken: String?,
```

Comment:
since this is kotlin and we can use defaults, and the old authorize is deprecated, can we add default nulls for these last 3 params?

Reply:
Since this is the interface I'll do it in the impl

Reply:
JK I can only do it here in the interface